### PR TITLE
Fix delegate properties - Closes #550

### LINKED ISSUE #550: Fix delegate properties
### Actual behavior

The `lastForgedHeight` property returns delegate registration height if there are no forged blocks by this delegate.

### Expected behavior

1. Make it distinguish the `lastForgedHeight` from the registration height.
2. In case the delegate never forged any block the `lastForgedHeight` has to be `null`.
3. Add the `registrationHeight`property to the API forging summary.
4. Make `producedBlocks` and `rewards` when the account did never forge.

```jsonc
{
    "data": [
        {
            "dpos": {
                "delegate": {
                    "consecutiveMissedBlocks": 0,
                    "isBanned": false,
                    "lastForgedHeight": 984987, // this should be null
                    "producedBlocks": null, // this should be 0
                    "rewards": null,         // this should be 0
                    "registrationHeight": 984987, // new property
                    "totalVotesReceived": "0",
                    "username": "sri"
                },
...
```

### Steps to reproduce

Call the URL
https://betanet5-service.lisk.io/api/v2/accounts?username=sri

### Which version(s) does this affect? (Environment, OS, etc...)

`0.3.0`

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?
This PR resolves #550 

### How was it solved?
- [x] `producedBlocks` is set to `0` for non-forging delegate accounts
- [x]  `rewards` is set to `0` for non-forging delegate accounts
- [x] `registrationHeight` is now returned for delegate accounts
- [x] `lastForgedHeight` is set to `null` for non-forging delegates
- [x] `lastForgedHeight` is set based on the `blocks` index
- [x] `registrationHeight` is returned only after the complete blockchain is successfully indexed
- [x] `registrationHeight` for a delegate defaults to the genesis block height, if no delegate registration transaction is available in the index
- [x] Set higher preference for a direct `targetPair` match from the source prices over calculated rates

### How was it tested?
Local + Jenkins

### CODE REVIEW

**Review by MichalTuleja — CHANGES_REQUESTED** [review]
Looks good but forging delegates resolve to `"registrationHeight": null`.
Is it possible to make this property filled for forging delegates?

**Comment by sameersubudhi** [comment]
> Looks good but forging delegates resolve to `"registrationHeight": null`.
> Is it possible to make this property filled for forging delegates?

`registrationHeight` is returned for all the delegates, once all the transaction index is synced completely. It returns `null` until the delegate registration transaction is not indexed. Alternatively, I could return the property only after the indexes are fully synchronized.

File: services/core/shared/core/compat/sdk_v5/accounts.js [hunk 1]
```diff
@@ -176,6 +175,32 @@ const resolveDelegateInfo = async accounts => {
...
 					.sort((a, b) => b - a).slice(0, 5)
 					.map(height => ({ start: height, end: height + punishmentHeight }));
+
+				const [lastForgedBlock = {}] = await blocksDB.find({
+					generatorPublicKey: account.publicKey,
+					limit: 1,
+				});
+				account.dpos.delegate.lastForgedHeight = lastForgedBlock.height || null;
+
+				// Iff the COMPLETE blockchain is SUCCESSFULLY indexed
```
Comment (nagdahimanshu) [inline services/core/shared/core/compat/sdk_v5/accounts.js:185]:
```suggestion
				// If the COMPLETE blockchain is SUCCESSFULLY indexed
```

**Review by nagdahimanshu — APPROVED** [review]
Looks good apart from one typo 👍

File: services/core/shared/core/compat/sdk_v5/accounts.js (hunk 1 above)
Comment (sameersubudhi) [reply services/core/shared/core/compat/sdk_v5/accounts.js:185]: @nagdahimanshu It is not a typo.`iff` is the acronym for `if and only if`.

Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/accounts.js:185]: Thanks for clarification 👍

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**